.github/workflows/knative-go-test.yaml

```diff
@@ -50,10 +50,11 @@ jobs:
           fetch-depth: 0
 
       - name: Merge upstream
+        if: github.event_name == 'pull_request'
         run: |
           git remote add upstream https://github.com/${{ github.repository }}.git
-          git fetch upstream ${{ github.base_ref == null && github.ref || github.base_ref }}
-          git pull --no-rebase --no-commit upstream ${{ github.base_ref == null && github.ref || github.base_ref }}
+          git fetch upstream ${{ github.base_ref }}
```

Comment:
@markusthoemmes this is a bug. When this action runs on `main` branch, `github.base_ref` is empty, failing the action. That's why I've changed.